repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/apivalidator/jsonschemacheck/JsonSchemaCheckTest.groovy

```diff
@@ -0,0 +1,111 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.filters.apivalidator.jsonschemacheck
+
+import framework.ReposeValveTest
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+import org.rackspace.deproxy.Response
+
+/**
+ * Created by jennyvo on 10/2/15.
+ */
+class JsonSchemaCheckTest extends ReposeValveTest {
+
+    def setupSpec() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+
+        def params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/apivalidator/jsonschemacheck", params)
+        repose.start()
+        repose.waitForNon500FromUrl(reposeEndpoint)
+    }
+
+    def cleanupSpec() {
+        if (repose)
+            repose.stop()
+        if (deproxy)
+            deproxy.shutdown()
+    }
+
+    def "GET on /path/to/test Json checking"() {
+        setup: "declare messageChain to be of type MessageChain"
+        MessageChain messageChain
+        def customHandler = { return new Response(200, "OK") }
+
+        def Map<String, String> headers = [
+                "Accept"         : "application/json",
+                "Content-Type"   : "application/json",
+                "Host"           : "localhost",
+                "User-Agent"     : "gdeproxy"
+        ]
+
+        def reqBody = """{
+            "firstName" : "Jorge",
+            "lastName" : "Williams",
+            "age" : 38
+            }"""
+
+        when:
+        messageChain = deproxy.makeRequest(url: reposeEndpoint + "/path/to/test",
+                method: 'GET', headers: headers,
+                requestBody: reqBody, defaultHandler: customHandler,
+                addDefaultHeaders: false
+        )
+
+        then:
+        "result should be " + 200
+        messageChain.receivedResponse.code.equals("200")
+        messageChain.receivedResponse.headers["Content-Type"].equals("application/json")
+    }
+
+    def "GET on /path/to/test Json checking Invalid Json"() {
+        setup: "declare messageChain to be of type MessageChain"
+        MessageChain messageChain
+        def customHandler = { return new Response(200, "OK") }
+
+        def Map<String, String> headers = [
+                "Accept"         : "application/json",
+                "Content-Type"   : "application/json",
+                "Host"           : "localhost",
+                "User-Agent"     : "gdeproxy"
+        ]
+
+        def reqBody = """{
+            "firstName" : "Jorge",
+            "lastName" : "Williams",
+            "age" : 38
+            }}"""
```

Comment:
The extra closing curly only makes this *malformed* JSON. To make it *invalid* it needs one of the required fields to be missing (misspelled) or to have an extra field that is not defined in the schema.